src/utils/dateRange.ts

```diff
@@ -51,3 +51,20 @@ export function getSinceDateRangeString(key: string = 'since_date') {
     startDate,
   });
 }
+
+export function getMonthDate(offset: number) {
+  const today = new Date();
+  today.setMonth(today.getMonth() - offset);
+  const startDate = new Date(today.getFullYear(), today.getMonth(), 1);
+  const endDate = offset === 0 ? new Date() : new Date(today.getFullYear(), today.getMonth() + 1, 0);
```

Comment:
This can be simplified to:

```
  const today = new Date();
  today.setMonth(today.getMonth() - offset);

  const start_date = format(startOfMonth(today), 'yyyy-MM-dd');
  const end_date = format(endOfMonth(today), 'yyyy-MM-dd');
```